src/Glpi/Database/AbstractDatabase.php

```diff
@@ -732,6 +767,47 @@ public function fieldExists(string $table, string $field, bool $usecache = true)
         return false;
     }
 
+    /**
+     * Check if an index exists
+     *
+     * @since 10.0
+     *
+     * @param string       $table    Table name for the field we're looking for
+     * @param string|array $field    Field(s) name(s)
+     * @param string       $name     Index name
+     * @param Boolean      $usecache Use cache; @see Database::listIndexes(), defaults to true
+     *
+     * @return boolean
+     */
+    public function indexExists(string $table, $field, $name = null, bool $usecache = true): bool
+    {
+
+        if (!$this->tableExists($table)) {
+            trigger_error("Table $table does not exists", E_USER_WARNING);
+            return false;
+        }
+
+        if (!is_array($field)) {
+            $field = [$field];
+        }
+        if ($indexes = $this->listIndexes($table, $usecache)) {
+            foreach ($indexes as $key => $current) {
+                if (null !== $name && $key == $name) {
+                    //an index with the name already exists
+                    return true;
+                }
+                sort($current);
```

Comment:
Not sure we have to sort.

Column order in compound index is important. I do not know if, on a really specific case, it could be usefull to have two indexes based on same columns but not same order (i.e. 'A+B' and 'B+A'). Having this kind of indexes looks more due to a bad DB/app design than a really necessary feature, but I think we should not change the input order.

In this case, we have to sort `listIndexes` query results using `SEQ_IN_INDEX` column.

Reply:
Well, it's difficult to know if requested order is the one used at the creation; that's why I've added the sort.

Reply:
Thinking about that, thins seems finally a bad idea. Expet from automatic usage from Migration, the common usage is to use a string; this seems simplier to keep a string.

Reply:
ok